delivery_state/models/stock_picking.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2020 Trey, Kilobytes de Soluciones
+# Copyright 2020 FactorLibre
+# Copyright 2020 Tecnativa - David Vidal
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    date_shipped = fields.Date(
+        string="Shipment Date",
+        readonly=True,
+    )
+    date_delivered = fields.Datetime(
+        string="Delivery Date",
+        readonly=True,
+    )
+    tracking_state = fields.Char(
+        readonly=True,
+        index=True,
+        tracking=True,
+    )
+    tracking_state_history = fields.Text(
+        readonly=True,
+    )
+    delivery_state = fields.Selection(
+        selection=[
+            ("shipping_recorded_in_carrier", "Shipping recorded in carrier"),
+            ("in_transit", "In transit"),
+            ("canceled_shipment", "Canceled shipment"),
+            ("incidence", "Incidence"),
```

Comment:
I think you wanted to say 'Incident' instead

Reply:
Oops you're right. An spanish false friend.

Reply:
Fixed!